plugins/jms-service-activemq/src/main/java/com/thinkbiganalytics/jms/activemq/ActiveMqConfig.java

```diff
@@ -45,50 +45,51 @@
-        factory.getRedeliveryPolicy().setUseExponentialBackOff(env.getProperty("jms.useExponentialBackOff", Boolean.class, false));
-        PooledConnectionFactory pool = new PooledConnectionFactory();
-        pool.setIdleTimeout(0);
-        pool.setConnectionFactory(getCredentialsAdapter(factory));
+	}
 
-        log.info("Setup ActiveMQ ConnectionFactory for " + env.getProperty("jms.activemq.broker.url"));
-        return pool;
-    }
+	@Bean (name = "activemqConnectionPool")
+	public ConnectionFactory connectionFactory() {
+		ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory(env.getProperty("jms.activemq.broker.url"));
+		factory.setTrustAllPackages(true);
+		factory.setRedeliveryPolicy(new RedeliveryPolicy());
+		factory.getRedeliveryPolicy().setMaximumRedeliveries(env.getProperty("jms.maximumRedeliveries", Integer.class, 100));
+		factory.getRedeliveryPolicy().setRedeliveryDelay(env.getProperty("jms.redeliveryDelay", Long.class, 1000L));
+		factory.getRedeliveryPolicy().setInitialRedeliveryDelay(env.getProperty("jms.initialRedeliveryDelay",Long.class,1000L));
+		factory.getRedeliveryPolicy().setMaximumRedeliveryDelay(env.getProperty("jms.maximumRedeliveryDelay", Long.class, 600000L));  // try for 10 min
+		factory.getRedeliveryPolicy().setBackOffMultiplier(env.getProperty("jms.backOffMultiplier", Double.class, 5d));
+		factory.getRedeliveryPolicy().setUseExponentialBackOff(env.getProperty("jms.useExponentialBackOff", Boolean.class, false));
+		PooledConnectionFactory pool = new PooledConnectionFactory();
+		pool.setIdleTimeout(0);
+		pool.setConnectionFactory(getCredentialsAdapter(factory));
 
-    private UserCredentialsConnectionFactoryAdapter getCredentialsAdapter(ConnectionFactory connectionFactory) {
-        UserCredentialsConnectionFactoryAdapter adapter = new UserCredentialsConnectionFactoryAdapter();
-        adapter.setTargetConnectionFactory(connectionFactory);
-        String username = env.getProperty("jms.activemq.broker.username");
-        String password = env.getProperty("jms.activemq.broker.password");
-        adapter.setUsername(username);
-        adapter.setPassword(password);
+		log.info("Setup ActiveMQ ConnectionFactory for " + env.getProperty("jms.activemq.broker.url"));
+		return pool;
+	}
 
-        log.info("Connecting to ActiveMQ {} ", username != null ? "as " + username : "anonymously");
+	private UserCredentialsConnectionFactoryAdapter getCredentialsAdapter(ConnectionFactory connectionFactory) {
+		UserCredentialsConnectionFactoryAdapter adapter = new UserCredentialsConnectionFactoryAdapter();
+		adapter.setTargetConnectionFactory(connectionFactory);
+		String username = env.getProperty("jms.activemq.broker.username");
+		String password = env.getProperty("jms.activemq.broker.password");
+		adapter.setUsername(username);
+		adapter.setPassword(password);
 
-        return adapter;
-    }
+		log.info("Connecting to ActiveMQ {} ", username != null ? "as " + username : "anonymously");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)